ttnn/cpp/ttnn/tensor/tensor_utils.cpp

```diff
@@ -473,10 +474,20 @@ const ttnn::SimpleShape infer_dims_for_reshape(const Tensor& tensor, tt::stl::Sp
             }
             index_of_negative_1 = index;
         } else {
-            TT_FATAL(shape[index] > 0, "New shape entries can only have -1 or positive values");
+            if (shape[index] == 0) {
+                has_zero = true;
+            }
             new_volume *= shape[index];
         }
     }
+    if (has_zero && index_of_negative_1 != -1) {
+        std::string error_msg = "cannot reshape tensor of 0 elements into shape [";
+        for(auto & s: shape) {
+            error_msg += std::to_string(s) + ",";
+        }
+        error_msg += "] because the unspecified dimension size -1 can be any value and is ambiguous";
+        TT_THROW("{}", error_msg);
```

Comment:
TT_THROW("Cannot reshape tensor of 0 elements into shape () because the unspecified dimension size -1 can be any value and is ambiguous", shape);

Reply:
cannot convert `shape` to string directly, so I remain the origin code

Reply:
What error do you get?

Reply:
The compiled err is
```
/home/ubuntu/tt-metal/.cpmcache/fmt/73b5ec45edbd92babfd91c3777a9e1ab9cac8238/include/fmt/base.h:2657:45: error: implicit instantiation of undefined template 'fmt::detail::type_is_unformattable_for<tt::stl::Span<const int>, char>'
 2657 |     type_is_unformattable_for<T, char_type> _;
```

Reply:
@ayerofieiev-tt `shape` here is `tt::stl::Span<const int32_t>`, not `ttnn::SimpleShape`, so it can't be easily printed

Reply:
I would suggest to extract error message printing code into a lambda and call it twice

Reply:
Can we pleace accept a SimpleShape here instead?

Reply:
Still like a lot of complication to me. 
We could attempt to create a `new_shape = ttnn::SimpleShape(std::move(new_shape));`
We could ask `new_shape.volume`, we could check things within it, instead of doing all the shenanigans to validate and print. wdyt?

Reply:
We can't create a SimpleShape out of a vector here. SimpleShape contains uint32_t, but the input numbers are int32_t with a special treatment of -1. The whole reason for this function is to replace -1 with a proper value.